tools/nnyeah/nnyeah/Program.cs

```diff
@@ -1,23 +1,74 @@
 using System;
 using System.IO;
+using System.Reflection;
+using Mono.Options;
+using System.Collections.Generic;
 
 #nullable enable
 
 namespace nnyeah {
     class Program {
         static void Main (string [] args)
         {
-            if (args.Length != 2) {
-                Console.Error.WriteLine ("Usage: dotnet nnyeah.dll /path/to/input/file.dll /path/to/output/file.dll");
-                Environment.Exit (1);
-            }
-            using var stm = new FileStream (args [0], FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+            var doHelp = false;
+            string? infile = null, outfile = null;
+            var verbose = false;
+            var suppressWarnings = false;
+            var warnings = new List<string> ();
+            var transforms = new List<string> ();
+
+            var options = new OptionSet () {
+                { "h|?|help", o => doHelp = true },
+                { "i=|input=", f => infile = f },
+                { "o=|output=", f => outfile = f },
+                { "v|verbose", o => verbose = true },
+                { "s|suppress-warnings", o => suppressWarnings = true },
+            };
+
+            var extra = options.Parse (args);
+
+            if (doHelp || infile is null || outfile is null) {
```

Comment:
options.Parse throws an exception, we should catch and the the appropriate thing. This is documented in most of the samples given https://github.com/xamarin/XamarinComponents/blob/main/XPlat/Mono.Options/samples/OptionsSample/Program.cs#L25
As the samples shown:
```csharp
List<string> extra;
			try {
				extra = p.Parse (args);
			} catch (OptionException e) {
				Console.Write ("OptionsSample.exe: ");
				Console.WriteLine (e.Message);
				Console.WriteLine ("Try `OptionsSample.exe --help' for more information.");
				return;
			}
```